test/test_integrate.py

```diff
@@ -146,13 +144,16 @@ def f(t, y, args):
 
 
 @pytest.mark.parametrize("solver", all_ode_solvers + all_split_solvers)
-def test_ode_order(solver):
+@pytest.mark.parametrize("dtype", [jnp.float64, jnp.complex128])
+def test_ode_order(solver, dtype):
     solver = implicit_tol(solver)
     key = jr.PRNGKey(5678)
     akey, ykey = jr.split(key, 2)
 
-    A = jr.normal(akey, (10, 10), dtype=jnp.float64) * 0.5
+    A = jr.normal(akey, (10, 10), dtype=dtype) * 0.5
 
+    if jnp.iscomplexobj(A) and isinstance(solver, diffrax.AbstractImplicitSolver):
+        return
```

Comment:
What's the reason for this exclusion? I think ideally we do expect to be able to do this (once all the relevant upstream things have landed).

Reply:
Currently,
https://github.com/patrick-kidger/diffrax/blob/main/diffrax/_integrate.py#L638-L642
I agree that we will want to remove it in the future

Reply:
It used to work (somehow) when we merged the basic support, but now I think strict typing broke it in multiple places.

Reply:
I've added some related fixes (also see https://github.com/patrick-kidger/optimistix/pull/53), but I think enabling it should come in a separate PR

Reply:
Okay, sounds reasonable!